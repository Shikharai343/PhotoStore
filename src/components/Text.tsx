import { Text as RNText, TextStyle } from 'react-native'
import React, { Children, FC, ReactElement } from 'react'

export type TextType = 'label' | 'title';
export type FontType = 'small' | 'regular';

const DEFAULT_FONT_FAMILY = 'Times New Roman'

const fontSize = {
    label: {
        small: 10,
        regular: 12,
    },
    title: {
        small: 16,
        regular: 18,
    },
};

export interface ITextProps {
    children: string | number;
    textType: TextType;
    fontType: FontType;
    color?: string;
    style?: TextStyle;
}

const Text: FC<ITextProps> = (props) => {
    // @ts-ignore
    const {textType, fontType, color, style, children} = props;

    const getTextStyles = (): TextStyle => {
        return {
            ...(style || {}),
            fontSize: fontSize[textType][fontType],
            fontFamily: DEFAULT_FONT_FAMILY,
            color
        }
    }
    return (
        <RNText style={getTextStyles()}>{children}</RNText>
    )
}

export default Text