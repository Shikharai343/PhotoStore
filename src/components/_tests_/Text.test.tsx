import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import Text, { FontType, ITextProps, TextType } from "../Text";

describe('Text', () => {
    const props: ITextProps = {
      children: 'Test',
      textType: 'label',
      fontType: 'small'
    };  

    const wrapper: ShallowWrapper<ITextProps, {}, Text> = shallow(
        <Text {...props} />
      );
    
      ['label' as TextType, 'title' as TextType].forEach((item: TextType) => {
        it(`should match snapshot for ${item}`, () => {
          wrapper.setProps({ ...props, textType: item });
          expect(toJson(wrapper)).toMatchSnapshot();
        });
      });
      ['small' as FontType, 'regular' as FontType].forEach((item: FontType) => {
        it(`should match snapshot for ${item}`, () => {
          wrapper.setProps({ ...props, fontType: item });
          expect(toJson(wrapper)).toMatchSnapshot();
        });
      });
  });