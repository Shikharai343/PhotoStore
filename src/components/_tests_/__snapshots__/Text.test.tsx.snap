// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot for label 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontFamily": "Times New Roman",
      "fontSize": 10,
    }
  }
>
  Test
</Text>
`;

exports[`Text should match snapshot for regular 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontFamily": "Times New Roman",
      "fontSize": 12,
    }
  }
>
  Test
</Text>
`;

exports[`Text should match snapshot for small 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontFamily": "Times New Roman",
      "fontSize": 10,
    }
  }
>
  Test
</Text>
`;

exports[`Text should match snapshot for title 1`] = `
<Text
  style={
    Object {
      "color": undefined,
      "fontFamily": "Times New Roman",
      "fontSize": 16,
    }
  }
>
  Test
</Text>
`;
